### ISSUE TITLE
base64Encode padding error

### ISSUE THREAD

**Issue by dshein-alt** [ClickHouse/ClickHouse#30854]
> You have to provide the following information whenever possible.

base64Encode function returns data with garbage instead of padding ('=') symbol.

> A clear and concise description of what works not as it is supposed to.

Text files contents stored in table in String type field.
Our Pyhton application reads it from CH trough clickhouse-driver with SQL request like:

`SELECT file_name, file_version, file_mtime, base64Encode(file_content) FROM Files where file_hash = 12345`

base64 encoding is used to get raw file contents not encoded in utf-8 charset.

When file contents selected directly from clickhouse-client it is displayed the same as in original file, but when encode to base64 on CH side it is decoded with some garbage bytes at the end depending on file contents.

I've found that CH base64Encode() function returns erroneous symbols instead of padding '=' ones. 

[The list of releases](https://github.com/ClickHouse/ClickHouse/blob/master/utils/list-versions/version_date.tsv)

**Enable crash reporting**

> If possible, change "enabled" to true in "send_crash_reports" section in `config.xml`:

```
<send_crash_reports>
        <!-- Changing <enabled> to true allows sending crash reports to -->
        <!-- the ClickHouse core developers team via Sentry https://sentry.io -->
        <enabled>false</enabled>
```

**How to reproduce**

* Which ClickHouse server version to use
* Which interface to use, if matters
* Non-default settings, if any
* `CREATE TABLE` statements for all tables involved
* Sample data for all these tables, use [clickhouse-obfuscator](https://github.com/ClickHouse/ClickHouse/blob/master/programs/obfuscator/Obfuscator.cpp#L42-L80) if necessary
* Queries to run that lead to unexpected result

Reproduced with official docker images:
```
    ClickHouse server version 21.8.10 revision 54449
    ClickHouse client version 21.10.2.15 (official build)
```

Reproducible by using clickhouse-client CLI and Python clickhouse-driver.

```
CREATE TABLE Files
(
    file_hash UInt64,
    file_name String,
    file_version String,
    file_mtime DateTime,
    file_content String
)
ENGINE ReplacingMergeTree
ORDER BY (file_hash, file_content);
```

Reproduced using latest Docker images with below contents:

```
SELECT base64Encode(base64Decode('JWRlZmluZSBfdW5wYWNrYWdlZF9maWxlc190ZXJtaW5hdGVfYnVpbGQgMQolZGVmaW5lIHNuYW1lIE5ldC1BTVFQCgpOYW1lOiBwZXJsLU5ldC1BTVFQClZlcnNpb246IDAuMDYKUmVsZWFzZTogYWx0MQpTdW1tYXJ5OiBBZHZhbmNlZCBNZXNzYWdlIFF1ZXVlIFByb3RvY29sIChkZSlzZXJpYWxpemF0aW9uIGFuZCByZXByZXNlbnRhdGlvbgpHcm91cDogRGV2ZWxvcG1lbnQvUGVybApMaWNlbnNlOiBwZXJsClVybDogaHR0cHM6Ly9tZXRhY3Bhbi5vcmcvcmVsZWFzZS9OZXQtQU1RUApTb3VyY2U6ICVzbmFtZS0ldmVyc2lvbi50YXIKQnVpbGRBcmNoOiBub2FyY2gKCkJ1aWxkUmVxdWlyZXM6IHBlcmwoQmVuY2htYXJrLnBtKSBwZXJsKENsYXNzL0FjY2Vzc29yL0Zhc3QucG0pIHBlcmwoQ2xhc3MvRGF0YS9Jbmhlcml0YWJsZS5wbSkgcGVybChFeHBvcnRlci5wbSkgcGVybChGaW5kQmluLnBtKSBwZXJsKE1vZHVsZS9CdWlsZC5wbSkgcGVybChTY2FsYXIvVXRpbC5wbSkgcGVybChUZXN0L0RlZXAucG0pIHBlcmwoVGVzdC9Nb3JlLnBtKSBwZXJsKFhNTC9MaWJYTUwucG0pIHBlcmwob3ZlcmxvYWQucG0pCgolZGVzY3JpcHRpb24KVGhpcyBtb2R1bGUgaW1wbGVtZW50cyB0aGUgZnJhbWUgKGRlKXNlcmlhbGl6YXRpb24gYW5kIHJlcHJlc2VudGF0aW9uIG9mCnRoZSBBZHZhbmNlZCBNZXNzYWdlIFF1ZXVlIFByb3RvY29sIChodHRwOi8vd3d3LmFtcXAub3JnLykuIEl0IGlzIHRvIGJlCnVzZWQgaW4gY29uanVuY3Rpb24gd2l0aCBjbGllbnQgb3Igc2VydmVyIHNvZnR3YXJlIHRoYXQgZG9lcyB0aGUgYWN0dWFsClRDUC9JUCBjb21tdW5pY2F0aW9uLgoKJXByZXAKJXNldHVwIC1xIC1uICVzbmFtZS0ldmVyc2lvbgoKJWJ1aWxkCiVwZXJsX3ZlbmRvcl9idWlsZAoKJWluc3RhbGwKJXBlcmxfdmVuZG9yX2luc3RhbGwKCiVmaWxlcwolZG9jIFJFQURNRSBMSUNFTlNFIENIQU5HRVMKJXBlcmxfdmVuZG9yX3ByaXZsaWIvTioKCiVjaGFuZ2Vsb2cKKiBXZWQgSnVsIDMxIDIwMTkgQWxleGFuZHIgQW50b25vdiA8YWFzQGFsdGxpbnV4Lm9yZz4gMC4wNi1hbHQxCi0gaW5pdGlhbCBidWlsZCBmb3IgQUxUCg=='))
```

Result is:
```
JWRlZmluZSBfdW5wYWNrYWdlZF9maWxlc190ZXJtaW5hdGVfYnVpbGQgMQolZGVmaW5lIHNuYW1lIE5ldC1BTVFQCgpOYW1lOiBwZXJsLU5ldC1BTVFQClZlcnNpb246IDAuMDYKUmVsZWFzZTogYWx0MQpTdW1tYXJ5OiBBZHZhbmNlZCBNZXNzYWdlIFF1ZXVlIFByb3RvY29sIChkZSlzZXJpYWxpemF0aW9uIGFuZCByZXByZXNlbnRhdGlvbgpHcm91cDogRGV2ZWxvcG1lbnQvUGVybApMaWNlbnNlOiBwZXJsClVybDogaHR0cHM6Ly9tZXRhY3Bhbi5vcmcvcmVsZWFzZS9OZXQtQU1RUApTb3VyY2U6ICVzbmFtZS0ldmVyc2lvbi50YXIKQnVpbGRBcmNoOiBub2FyY2gKCkJ1aWxkUmVxdWlyZXM6IHBlcmwoQmVuY2htYXJrLnBtKSBwZXJsKENsYXNzL0FjY2Vzc29yL0Zhc3QucG0pIHBlcmwoQ2xhc3MvRGF0YS9Jbmhlcml0YWJsZS5wbSkgcGVybChFeHBvcnRlci5wbSkgcGVybChGaW5kQmluLnBtKSBwZXJsKE1vZHVsZS9CdWlsZC5wbSkgcGVybChTY2FsYXIvVXRpbC5wbSkgcGVybChUZXN0L0RlZXAucG0pIHBlcmwoVGVzdC9Nb3JlLnBtKSBwZXJsKFhNTC9MaWJYTUwucG0pIHBlcmwob3ZlcmxvYWQucG0pCgolZGVzY3JpcHRpb24KVGhpcyBtb2R1bGUgaW1wbGVtZW50cyB0aGUgZnJhbWUgKGRlKXNlcmlhbGl6YXRpb24gYW5kIHJlcHJlc2VudGF0aW9uIG9mCnRoZSBBZHZhbmNlZCBNZXNzYWdlIFF1ZXVlIFByb3RvY29sIChodHRwOi8vd3d3LmFtcXAub3JnLykuIEl0IGlzIHRvIGJlCnVzZWQgaW4gY29uanVuY3Rpb24gd2l0aCBjbGllbnQgb3Igc2VydmVyIHNvZnR3YXJlIHRoYXQgZG9lcyB0aGUgYWN0dWFsClRDUC9JUCBjb21tdW5pY2F0aW9uLgoKJXByZXAKJXNldHVwIC1xIC1uICVzbmFtZS0ldmVyc2lvbgoKJWJ1aWxkCiVwZXJsX3ZlbmRvcl9idWlsZAoKJWluc3RhbGwKJXBlcmxfdmVuZG9yX2luc3RhbGwKCiVmaWxlcwolZG9jIFJFQURNRSBMSUNFTlNFIENIQU5HRVMKJXBlcmxfdmVuZG9yX3ByaXZsaWIvTioKCiVjaGFuZ2Vsb2cKKiBXZWQgSnVsIDMxIDIwMTkgQWxleGFuZHIgQW50b25vdiA8YWFzQGFsdGxpbnV4Lm9yZz4gMC4wNi1hbHQxCi0gaW5pdGlhbCBidWlsZCBmb3IgQUxUCgCl
```

**Expected behavior**

> A clear and concise description of what you expected to happen.

Expects that base64Encode() returns consistent data representation in accordance to standard.

**Error message and/or stacktrace**

> If applicable, add screenshots to help explain your problem.

**Additional context**

> Add any other context about the problem here.

**Comment by alexey-milovidov** [comment]
#31643

**Comment by den-crane** [comment]
fixed  #31797

```sql
21.13.1.9141

SELECT substring(hex(base64Decode(base64Encode(repeat('a', 46)))), -10) AS r
┌─r──────────┐
│ 6161616161 │
└────────────┘


SELECT
    number,
    substring(hex(base64Decode(base64Encode(repeat('a', number)))), -10) AS r
FROM numbers(100)

┌─number─┬─r──────────┐
│      0 │            │
│      1 │ 61         │
│      2 │ 6161       │
│      3 │ 616161     │
│      4 │ 61616161   │
│      5 │ 6161616161 │
│      6 │ 6161616161 │
│      7 │ 6161616161 │
│      8 │ 6161616161 │
│      9 │ 6161616161 │
│     10 │ 6161616161 │
│     11 │ 6161616161 │
│     12 │ 6161616161 │
│     13 │ 6161616161 │
│     14 │ 6161616161 │
│     15 │ 6161616161 │
│     16 │ 6161616161 │
│     17 │ 6161616161 │
│     18 │ 6161616161 │
│     19 │ 6161616161 │
│     20 │ 6161616161 │
│     21 │ 6161616161 │
│     22 │ 6161616161 │
│     23 │ 6161616161 │
│     24 │ 6161616161 │
│     25 │ 6161616161 │
│     26 │ 6161616161 │
│     27 │ 6161616161 │
│     28 │ 6161616161 │
│     29 │ 6161616161 │
│     30 │ 6161616161 │
│     31 │ 6161616161 │
│     32 │ 6161616161 │
│     33 │ 6161616161 │
│     34 │ 6161616161 │
│     35 │ 6161616161 │
│     36 │ 6161616161 │
│     37 │ 6161616161 │
│     38 │ 6161616161 │
│     39 │ 6161616161 │
│     40 │ 6161616161 │
│     41 │ 6161616161 │
│     42 │ 6161616161 │
│     43 │ 6161616161 │
│     44 │ 6161616161 │
│     45 │ 6161616161 │
│     46 │ 6161616161 │
│     47 │ 6161616161 │
│     48 │ 6161616161 │
│     49 │ 6161616161 │
│     50 │ 6161616161 │
│     51 │ 6161616161 │
│     52 │ 6161616161 │
│     53 │ 6161616161 │
│     54 │ 6161616161 │
│     55 │ 6161616161 │
│     56 │ 6161616161 │
│     57 │ 6161616161 │
│     58 │ 6161616161 │
│     59 │ 6161616161 │
│     60 │ 6161616161 │
│     61 │ 6161616161 │
│     62 │ 6161616161 │
│     63 │ 6161616161 │
│     64 │ 6161616161 │
│     65 │ 6161616161 │
│     66 │ 6161616161 │
│     67 │ 6161616161 │
│     68 │ 6161616161 │
│     69 │ 6161616161 │
│     70 │ 6161616161 │
│     71 │ 6161616161 │
│     72 │ 6161616161 │
│     73 │ 6161616161 │
│     74 │ 6161616161 │
│     75 │ 6161616161 │
│     76 │ 6161616161 │
│     77 │ 6161616161 │
│     78 │ 6161616161 │
│     79 │ 6161616161 │
│     80 │ 6161616161 │
│     81 │ 6161616161 │
│     82 │ 6161616161 │
│     83 │ 6161616161 │
│     84 │ 6161616161 │
│     85 │ 6161616161 │
│     86 │ 6161616161 │
│     87 │ 6161616161 │
│     88 │ 6161616161 │
│     89 │ 6161616161 │
│     90 │ 6161616161 │
│     91 │ 6161616161 │
│     92 │ 6161616161 │
│     93 │ 6161616161 │
│     94 │ 6161616161 │
│     95 │ 6161616161 │
│     96 │ 6161616161 │
│     97 │ 6161616161 │
│     98 │ 6161616161 │
│     99 │ 6161616161 │
└────────┴────────────┘
```
